File: 2019/scripts/python/day-01.py
input_data = [int(x) for x in open('2019/input/input-01.txt', 'r').read().split()]
    
# Part 1
get_fuel = lambda x: int(x)//3-2
print(sum(list(map(get_fuel, input_data))))

# Part 2
fuelception = lambda fuel: 0 if fuel <= 0 else fuel + fuelception(get_fuel(fuel))
print(sum(list(map(fuelception, map(get_fuel, input_data)))))

File: 2019/scripts/r/day-01.R
# --- Day 1: The Tyranny of the Rocket Equation ---

input <- scan("2019/input/input-01.txt")

# Part 1

sum(input %/% 3 - 2)

# Part 2

sum(sapply(input,
           function(x) {
             fuel <- x %/% 3 - 2
             if (fuel <= 0) {
               return(0)
             } else {
               return(fuel + Recall(fuel))
             }
           }))
